diff -rU5 qt-everywhere-opensource-src-4.8.7old/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp qt-everywhere-opensource-src-4.8.7/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp
--- qt-everywhere-opensource-src-4.8.7old/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp	2015-05-07 16:14:47.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.7/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp	2023-03-12 20:37:32.000000000 +0100
@@ -72,11 +72,11 @@
 PassRefPtr<HTMLImageElement> HTMLImageElement::createForJSConstructor(Document* document, const int* optionalWidth, const int* optionalHeight)
 {
     RefPtr<HTMLImageElement> image = adoptRef(new HTMLImageElement(imgTag, document));
     if (optionalWidth)
         image->setWidth(*optionalWidth);
-    if (optionalHeight > 0)
+    if (optionalHeight)
         image->setHeight(*optionalHeight);
     return image.release();
 }
 
 bool HTMLImageElement::mapToEntry(const QualifiedName& attrName, MappedAttributeEntry& result) const
diff -rU5 qt-everywhere-opensource-src-4.8.7old/tools/linguist/linguist/messagemodel.cpp qt-everywhere-opensource-src-4.8.7/tools/linguist/linguist/messagemodel.cpp
--- qt-everywhere-opensource-src-4.8.7old/tools/linguist/linguist/messagemodel.cpp	2015-05-07 16:14:39.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.7/tools/linguist/linguist/messagemodel.cpp	2023-03-13 22:20:31.000000000 +0100
@@ -181,11 +181,11 @@
     for (int i = 0; i < two->contextCount(); ++i) {
         ContextItem *oc = two->contextItem(i);
         if (ContextItem *c = one->findContext(oc->context())) {
             for (int j = 0; j < oc->messageCount(); ++j) {
                 MessageItem *m = oc->messageItem(j);
-                if (c->findMessage(m->text(), m->comment()) >= 0)
+                if (c->findMessage(m->text(), m->comment()))
                     ++inBoth;
             }
         }
     }
     return inBoth * 100 / two->messageCount();
