diff -rU5 tesseract-4.1.1old/src/textord/devanagari_processing.cpp tesseract-4.1.1/src/textord/devanagari_processing.cpp
--- tesseract-4.1.1old/src/textord/devanagari_processing.cpp	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1/src/textord/devanagari_processing.cpp	2024-08-19 17:42:51.000000000 +0200
@@ -20,16 +20,19 @@
 
 #ifdef HAVE_CONFIG_H
 #include "config_auto.h"
 #endif
 
-#include "allheaders.h"
-#include "debugpixa.h"
 #include "devanagari_processing.h"
+
+#include "debugpixa.h"
 #include "statistc.h"
 #include "tordmain.h"
 
+#include <allheaders.h>
+#include <pix_internal.h>
+
 // Flags controlling the debugging information for shiro-rekha splitting
 // strategies.
 INT_VAR(devanagari_split_debuglevel, 0,
         "Debug level for split shiro-rekha process.");
 
diff -rU5 tesseract-4.1.1old/src/textord/devanagari_processing.h tesseract-4.1.1/src/textord/devanagari_processing.h
--- tesseract-4.1.1old/src/textord/devanagari_processing.h	2019-12-26 15:21:51.000000000 +0100
+++ tesseract-4.1.1/src/textord/devanagari_processing.h	2024-08-19 18:14:02.000000000 +0200
@@ -11,11 +11,11 @@
 // limitations under the License.
 
 #ifndef TESSERACT_TEXTORD_DEVNAGARI_PROCESSING_H_
 #define TESSERACT_TEXTORD_DEVNAGARI_PROCESSING_H_
 
-#include "allheaders.h"
+#include <allheaders.h>
 #include "ocrblock.h"
 #include "params.h"
 
 struct Pix;
 struct Box;
@@ -31,10 +31,12 @@
 
 class TBOX;
 
 namespace tesseract {
 
+class DebugPixa;
+
 class PixelHistogram {
  public:
   PixelHistogram() {
     hist_ = nullptr;
     length_ = 0;
