diff -rU5 khangman-4.14.3old/src/desktop/khangmanview.cpp khangman-4.14.3/src/desktop/khangmanview.cpp
--- khangman-4.14.3old/src/desktop/khangmanview.cpp	2014-08-27 07:30:36.000000000 +0200
+++ khangman-4.14.3/src/desktop/khangmanview.cpp	2023-03-25 19:48:27.914405843 +0100
@@ -508,11 +508,11 @@
         popup->setTimeout( Prefs::missedTimer()*1000 );
         popup->setView(i18n("KHangMan"), i18n("This letter has already been guessed.") );
 
         int  x = 0;
         int  y = 0;
-        if (m_missedLetters.contains(guess) > 0) {
+        if (m_missedLetters.contains(guess)) {
             // FIXME: popup should be better placed.
 
             QPoint abspos = popup->pos();
             x = abspos.x() + width()*400/700;
             y = abspos.y() + height()*50/535;
@@ -525,11 +525,11 @@
 
             // Disable any possible entry
             m_letterInput->setEnabled(false);
         }
 
-        if (goodWord.contains(guess) > 0) {
+        if (goodWord.contains(guess)) {
             point = m_theme->goodWordPos(size(), popup->pos());
             
             QTimer::singleShot( Prefs::missedTimer() * 1000,
                                 this, SLOT(enableUserInput()) );
 
