--- audacity.SlackBuild	2020-07-01 18:56:55.000000000 +0200
+++ audacity.SlackBuild.new2	2020-07-01 19:22:17.000000000 +0200
@@ -298,7 +298,7 @@
 install -vDm 644 nyquist/*.lsp -t ${PKG}/usr/share/${PRGNAM}/nyquist
 
 # ldd complains if the shared libraries are not executable:
-chmod 0755 ${PKG}/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/*.so
+# chmod 0755 ${PKG}/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/*.so
 
 # Add this to the doinst.sh:
 mkdir -p $PKG/install
