--- audacity.SlackBuild.orig	2020-11-12 23:47:18.000000000 +0100
+++ audacity.SlackBuild	2020-11-12 23:52:42.000000000 +0100
@@ -288,10 +288,12 @@
     -Daudacity_use_soundtouch=local \
     -Daudacity_use_twolame=system \
     -Daudacity_use_vamp=system \
     -Daudacity_use_vorbis=system \
     -Daudacity_use_wxwidgets=local \
+    -DwxBUILD_OPTIMISE:BOOL=ON \
+    -DwxBUILD_SHARED:BOOL=OFF \
   2>&1 | tee $OUTPUT/configure-${PRGNAM}.log
 
   make $NUMJOBS 2>&1 | tee $OUTPUT/make-${PRGNAM}.log
   make DESTDIR=$PKG install 2>&1 | tee $OUTPUT/install-${PRGNAM}.log
 cd -
