--- chromium.SlackBuild.orig	2020-05-22 09:40:40.000000000 +0200
+++ chromium.SlackBuild	2020-05-25 23:23:58.000000000 +0200
@@ -1000,12 +1000,12 @@
   $PKG/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/
 if [ -f out/Release/libffmpeg*.so ]; then
   cp -a \
-    out/Release/libffmpeg*.so* \
+    out/Release/lib*.so* \
     $PKG/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/
 elif [ -f out/Release/lib/libffmpeg.so ]; then
   mkdir -p $PKG/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/lib/
   cp -a \
-    out/Release/lib/libffmpeg.so* \
+    out/Release/lib/lib*.so* \
     $PKG/usr/lib${LIBDIRSUFFIX}/${PRGNAM}/lib/
 fi
 for BLOB in out/Release/{natives_blob.bin,snapshot_blob.bin,v8_context_snapshot.bin};
