--- lilv.SlackBuild	2020-05-28 11:23:03.000000000 +0200
+++ lilv.SlackBuild.new	2020-05-28 11:25:46.000000000 +0200
@@ -41,7 +41,7 @@
 # -----------------------------------------------------------------------------
 
 PRGNAM=lilv
-VERSION=${VERSION:-0.24.4}
+VERSION=${VERSION:-0.24.8}
 BUILD=${BUILD:-1}
 NUMJOBS=${NUMJOBS:-" -j$(nproc) "}
 TAG=${TAG:-alien}
@@ -162,7 +162,6 @@
   --configdir=/etc \
   --mandir=/usr/man \
   --docdir=/usr/doc/$PRGNAM-$VERSION \
-  --bindings \
   --no-bash-completion \
   --dyn-manifest \
   2>&1 | tee $OUTPUT/configure-${PRGNAM}.log
