--- wxGTK3.SlackBuild.old	2021-01-09 17:11:24.000000000 +0100
+++ wxGTK3.SlackBuild	2021-01-09 17:12:03.000000000 +0100
@@ -179,10 +179,12 @@
   --enable-mediactrl \
   --enable-plugins \
   --enable-unicode \
   --with-gtk=3 \
   --with-opengl \
+  --disable-webkit \
+  --disable-webview \
   --program-prefix= \
   --program-suffix= \
   --build=$TARGET \
   2>&1 | tee $OUTPUT/configure-${PRGNAM}.log
 make $NUMJOBS 2>&1 | tee $OUTPUT/make-${PRGNAM}.log
