--- calibre-2.71.0/setup/build.py.orig	2016-10-31 04:57:22.000000000 +0100
+++ calibre-2.71.0/setup/build.py	2016-11-08 14:14:29.310536819 +0100
@@ -368,7 +368,7 @@
             INCLUDEPATH += {freetype}
             DESTDIR = {destdir}
             CONFIG -= create_cmake  # Prevent qmake from generating a cmake build file which it puts in the calibre src directory
-            QMAKE_LIBS_PRIVATE += {glib} {fontconfig}
+            QMAKE_LIBS_PRIVATE += {glib} {fontconfig} -lfreetype
             ''').format(
                 headers=' '.join(headers), sources=' '.join(sources), others=' '.join(others), destdir=self.d(
                     target), glib=glib_flags, fontconfig=fontconfig_flags, freetype=' '.join(ft_inc_dirs))
