diff -rU5 glib-2.70.3old/gio/tests/meson.build glib-2.70.3/gio/tests/meson.build
--- glib-2.70.3old/gio/tests/meson.build	2022-01-26 15:53:56.000000000 +0100
+++ glib-2.70.3/gio/tests/meson.build	2024-09-01 18:05:10.000000000 +0200
@@ -568,11 +568,11 @@
       configuration: cdata,
     )
   endforeach
 endif
 
-if not meson.is_cross_build() or meson.has_exe_wrapper()
+if not meson.is_cross_build()
 
   plugin_resources_c = custom_target('plugin-resources.c',
     input : 'test4.gresource.xml',
     output : 'plugin-resources.c',
     command : [glib_compile_resources,
